File: day1/part2.py
#!/usr/bin/env python3

import sys
fname = "input"
with open(fname, "rt") as file:
    lines = file.readlines()

numbers = [int(x) for x in lines]

total = 0
# for i in range(0, len(numbers)):
# 	print (i / len(numbers) * 100)
# 	for j in range(0, i):
# 		for k in range(0, j):
# 			total += 1
# 			ni = numbers[i]
# 			nj = numbers[j]
# 			nk = numbers[k]
# 			# print ("%s %s %s %s %s" % (ni, nj, nk, ni + nj + nk, ni*nj*nk))
# 			if (ni + nj + nk == 2020):
# 				print ("%s %s %s %s %s" % (ni, nj, nk, ni + nj + nk, ni*nj*nk))


for i in numbers:
	for j in numbers:
		for k in numbers:
			if (i + j + k == 2020):
				print ("%s %s %s %s %s" % (i, j, k, i + j + k, i*j*k))
print (total)

File: day13/part2.js
#!/usr/bin/env node

import { readLines } from '../common/readers.js';

async function main() {	

	const lines = await readLines('input');
	const buses = lines[1].split(',');
	let busData = [];
	for (let i = 0; i < buses.length; ++i) {
		if (buses[i] !== 'x') {
			busData.push({
				bus: +buses[i],
				order: i
			});
		}
	}
	busData.sort((a, b) => b.bus - a.bus);

	let noMatch = true;
	let increment = 1;
	// let ts = 1;
	let ts = 100000000000000;
	while(noMatch) {
		noMatch = false;
		console.log(`Timestamp: ${ts} Increment ${increment}`);
		let newIncrement = 1;
		for (const theBus of busData) {
			if (((ts + theBus.order) % theBus.bus) === 0) {
				console.log(`${theBus.bus} departs at ${ts} + ${theBus.order}`);
				newIncrement *= theBus.bus;
			}
			else {
				noMatch = true;
				break;
			}
		}
		increment = newIncrement;
		ts += increment;
	}
}

main();
